File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/staff/StaffDTO.java
package jp.co.worksap.intern.entities.staff;

import jp.co.worksap.intern.entities.ICsvMasterDTO;

public class StaffDTO implements ICsvMasterDTO {
	/**
	 * 
	 */
	private static final long serialVersionUID = -8538243149394207296L;
	
	private Long staffId;
	private String name;
	private String gender;
	private RankType rank;
	private PositionType position;
	private Long supermarketId;

	/**
	 * Default Constructor
	 * 
	 * @param staffId
	 * @param name
	 * @param gender
	 * @param rank
	 * @param position
	 * @param supermarketId
	 */
	public StaffDTO(Long staffId, String name, String gender,
			RankType rank, PositionType position, Long supermarketId) {
		super();
		this.staffId = staffId;
		this.name = name;
		this.gender = gender;
		this.rank = rank;
		this.position = position;
		this.supermarketId = supermarketId;
	}

	public Long getStaffId() {
		return staffId;
	}

	public String getName() {
		return name;
	}

	public String getGender() {
		return gender;
	}

	public RankType getRank() {
		return rank;
	}

	public PositionType getPosition() {
		return position;
	}

	public Long getSupermarketId() {
		return supermarketId;
	}

}

File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Cpp/src/entities/staff/staff_dto.cpp
#include "staff_dto.h"
#include "rank_type.h"
#include "position_type.h"
#include <iostream>
using namespace std;

StaffDTO::StaffDTO(long staff_id, string name, string gender,
		RankType rank, PositionType position, long shop_id) {
	this->staff_id = staff_id;
	this->name = name;
	this->gender = gender;
	this->rank = rank;
	this->position = position;
	this->supermarket_id = supermarket_id;
}

long &StaffDTO::get_staff_id() {
	return staff_id;
}

string &StaffDTO::get_name() {
	return name;
}

string &StaffDTO::get_gender() {
	return gender;
}

RankType &StaffDTO::get_rank() {
	return rank;
}

PositionType &StaffDTO::get_position() {
	return position;
}

long &StaffDTO::get_supermarket_id() {
	return supermarket_id;
}

ostream &operator <<(ostream &out, StaffDTO &obj) {
	out << obj.staff_id << " " << obj.name << " " << obj.gender << " "
			<< rank_type_to_string(obj.rank) << " "
			<< position_type_to_string(obj.position) << " " << obj.supermarket_id;
	return out;
}

istream &operator >>(istream &in, StaffDTO &obj) {
	string rank_type_str;
	string position_type_str;
	in >> obj.staff_id >> obj.name >> obj.gender >> rank_type_str
			>> position_type_str >> obj.supermarket_id;
	obj.rank = rank_type_value_of_string(rank_type_str);
	obj.position = position_type_value_of_string(position_type_str);
	return in;
}

